public/pages/ThreatIntel/containers/AddThreatIntelSource/AddThreatIntelSource.tsx

```diff
@@ -99,50 +158,155 @@ export const AddThreatIntelSource: React.FC<AddThreatIntelSourceProps> = ({
     });
   };
 
+  const validateSourceName = (name: string) => {
+    let error;
+    if (!name) {
+      error = 'Name is required.';
+    } else if (name.length > 128) {
+      error = 'Max length can be 128.';
+    } else {
+      const isValid = validateName(name, THREAT_INTEL_SOURCE_NAME_REGEX);
+      if (!isValid) {
+        error = 'Invalid name.';
+      }
+    }
+
+    return error;
+  };
   const onNameChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const name = event.target.value;
+    const nameError = validateSourceName(name);
+    setFieldError({
+      name: nameError || '',
+    });
+
     setSource({
       ...source,
-      name: event.target.value,
+      name,
     });
+    setFieldTouched({ name: true });
   };
 
+  const validateSourceDescription = (description: string) => {
+    let error;
+    const isValid = validateDescription(description, THREAT_INTEL_SOURCE_DESCRIPTION_REGEX);
+    if (!isValid) {
+      error = 'Invalid name.';
+    }
+
+    return error;
+  };
   const onDescriptionChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const descriptionError = validateSourceDescription(event.target.value);
+    setFieldError({
+      description: descriptionError || '',
+    });
+
     setSource({
       ...source,
       description: event.target.value,
     });
+    setFieldTouched({ description: true });
   };
 
+  const validateS3ConfigField = (value: string) => {
+    if (!value) {
+      return `Required.`;
+    }
+  };
   const onS3DataChange = (field: keyof S3ConnectionSource['s3'], value: string) => {
+    const error = validateS3ConfigField(value);
+    setFieldError({
+      s3: {
+        ...inputErrors.s3,
+        [field]: error || '',
+      },
+    });
     setS3ConnectionDetails({
       s3: {
         ...s3ConnectionDetails.s3,
         [field]: value,
       },
     });
+    setFieldTouched({ s3: { ...inputTouched.s3, [field]: true } });
   };
 
+  const validateSchedule = (schedule: PeriodSchedule) => {
+    setFieldError({
+      schedule:
+        !schedule.period.interval || Number.isNaN(schedule.period.interval)
+          ? 'Invalid schedule'
```

Comment:
Not blocking:
Should the error message end in a period?

Reply:
Fixed